decred/decred/dcr/dcrdata.py

```diff
@@ -941,7 +941,7 @@ def pubsubSignal(self, sig):
             if sigType == "address":
                 msg = sig["message"]
                 log.debug("signal received for %s" % msg["address"])
-                self.addressReceiver(sig)
+                self.addressReceiver(msg["address"], msg["transaction"])
```

Comment:
This was changed in #98

Reply:
Isn't that what i said in the PR description?

> Bug introduced in #98 that I didn't catch in testing.

My question is why was it changed and not tested? The only method [currently being assigned](https://github.com/decred/tinydecred/blob/39d3b506fcb90227560c8580e4cc6ca905c1eacb/decred/decred/dcr/account.py#L1988) to `addressReceiver` is `Account.addressSignal`, which [takes two arguments](https://github.com/decred/tinydecred/blob/39d3b506fcb90227560c8580e4cc6ca905c1eacb/decred/decred/dcr/account.py#L1793), address and transaction ID. Sending a transaction on master right now results in an error. 

```
dcrdata ERROR pubsubSignal(859) failed to process pubsub message: addressSignal() missing 1 required positional argument: 'txid'
```

Any of the following three actions will cause the error

1. Send some DCR
2. Buy a ticket
3. Run the example scripts

Reply:
Yes you did, sorry about that.

I don't see it if I don't send the dcr first, but I do see it now.

I also get a lot of output when buying tickets, when spentTickets tries to fetch data from dcrdata that isn't ready yet. Unrelated to this though.

Reply:
My bad, sorry. I made the change in #98 because the `subscribeAddresses` docstring [shows](https://github.com/decred/tinydecred/blob/master/decred/decred/dcr/dcrdata.py#L569) the `receiver` signature as accepting one `obj` argument, not two arbitrary ones. I made the change here and in tests but missed the usage pointed out here.

If reverting this change, the `subscribeAddresses` signature should also be updated and the tests fixed. Another fix is changing the `Account.addressSignal` to accept a single dict instead, with the two current arguments as items.

Reply:
> Another fix is changing the Account.addressSignal to accept a single dict instead, with the two current arguments as items.

The choice to pass the address and transaction ID separately was intentional. Ideally, the `Blockchain` class should be interchangeable with any other implementation. The `dict` parsed from dcrdata will not be the same as the notification from dcrd when the full-node `Blockchain` is implemented.

That said, the implementation of `DcrdataBlockchain` likely has some areas that are not source-independent, e.g. block notifications. One of the challenges of upcoming work will be to more clearly define what is the `Blockchain` API and what are implementation details.